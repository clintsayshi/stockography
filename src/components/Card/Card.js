import React from "react";
import { Link } from "react-router-dom";

function Card({ data }) {
  const { id, tags, previewHeight, previewWidth, previewURL } = data;
  return (
    <Link className="relative" to={`/${id}`} key={id}>
      <img
        className="w-full rounded-lg h-60 object-cover"
        alt={tags}
        width={previewWidth}
        height={previewHeight}
        src={previewURL}
      />
    </Link>
  );
}

export default Card;
