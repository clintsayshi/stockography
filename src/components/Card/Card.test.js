import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import Card from "./Card";
import { MemoryRouter } from "react-router-dom";

const fakeData = {
  id: 132132,
  tags: "nature, sunrise",
  previewHeight: 250,
  previewWidth: 200,
  previewURL:
    "https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg",
};

afterEach(() => {
  cleanup();
});

test("renders Card successfully", () => {
  render(
    <MemoryRouter>
      <Card data={fakeData} />
    </MemoryRouter>
  );

  const cardElement = screen.getByTestId("card-132132");

  expect(cardElement).toBeInTheDocument();
  expect(cardElement).toMatchSnapshot();
});

test("the link works as expected", () => {
  render(
    <MemoryRouter>
      <Card data={fakeData} />
    </MemoryRouter>
  );

  expect(screen.getByRole("link")).toHaveAttribute("href", "/132132");
  expect(screen.getByRole("img")).toHaveAttribute(
    "src",
    "https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg"
  );
  expect(screen.getByRole("img")).toHaveAttribute("alt", "nature, sunrise");
});
