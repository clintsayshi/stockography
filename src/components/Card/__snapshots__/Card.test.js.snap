// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card successfully 1`] = `
<a
  className="relative"
  data-testid="card-132132"
  href="/132132"
  onClick={[Function]}
>
  <img
    alt="nature, sunrise"
    className="w-full rounded-lg h-60 object-cover"
    height={250}
    src="https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg"
    width={200}
  />
</a>
`;
